### DISCUSSION TITLE
USBDebugger Resource supported vendors/models

### DISCUSSION THREAD

**Discussion by snerzkristall** [labgrid-project/labgrid#986] [Q&A] [answered]
As of now, only two vendors/models are supported.
Is it possible to use other USBDebuggers (e.g. Olimex ARM-USB-OCD-H) with the openocd driver?

Simply adding the vendor and model ID in /labgrid/resource/udev.py a shown below does not work.

```
class USBDebugger(USBResource):
    def filter_match(self, device):
        match = (device.properties.get('ID_VENDOR_ID'), device.properties.get('ID_MODEL_ID'))

        if match not in [
                         ("0403", "6010"),  # FT2232C/D/H Dual UART/FIFO IC
                         ("0483", "374f"),  # STLINK-V3
                         ("15ba", "002b"),  # Olimex ARM-USB-OCD-H
                         ]:
            return False

        return super().filter_match(device)
```

### EDIT Solution
- adding vendor/vodel ID to udev.py does work **after restarting**
- will create PR to add Olimex devices in future

**Comment by jluebbe** [comment, marked as answer]
If that device is supported by OpenOCD, adding the vendor/model IDs should be enough. Does it show up with `labgrid-suggest` after your addition?

**Reply by snerzkristall** [reply]
Flashing/Debugging STM32 devices directly is possible with openocd-0.10.0 (Raspi 4b --> STM32 Board).

Works now after restarting raspi... :)

Thanks a lot anyway!

**Reply by jluebbe** [reply]
We'd be happy to merge a PR which adds these IDs. :)
